presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRemoteSourceOperator.java

```diff
@@ -70,13 +69,12 @@ public Page getOutput()
             return null;
         }
 
-        synchronized (iterator) {
-            if (!iterator.hasNext()) {
-                finished = true;
-                return null;
-            }
-            return iterator.next();
+        Page page = pageInput.getNextPage();
```

Comment:
A nice refactor to have named `PrestoSparkPageInput` class (instead of a general iterator).

Curious why now syntonization is not required?

Reply:
Now the `pageInput` ensures synchronization. The `PrestoSparkPageInput` implementation is expected to be thread safe.